hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java

```diff
@@ -892,13 +918,164 @@ private static ConditionQuery matchIndexLabel(ConditionQuery query,
                     indexQuery.query(condition);
                 }
                 break;
+            case SHARD:
+                HugeType type = indexLabel.indexType().type();
+                indexQuery = new ConditionQuery(type, query);
+                indexQuery.eq(HugeKeys.INDEX_LABEL_ID, indexLabel.id());
+                List<Condition> conditions = constructShardConditions(
+                                             query, indexLabel.indexFields(),
+                                             HugeKeys.FIELD_VALUES);
+                for (Condition c : conditions) {
+                    indexQuery.query(c);
+                }
+                break;
             default:
                 throw new AssertionError(String.format(
                           "Unknown index type '%s'", indexType));
         }
         return indexQuery;
     }
 
+    public static List<Condition> constructShardConditions(ConditionQuery query,
+                                                           List<Id> fields,
+                                                           HugeKeys key) {
+        List<Condition> conditions = new ArrayList<>(2);
+        String joinedValues;
+        boolean hasRange = false;
+        int usedCondCount = 0;
+        int equalCondCount = 0;
+        List<Object> prefixes = new ArrayList<>();
+        for (Id field : fields) {
+            List<Condition> conds = query.userpropConditions(field);
+            if (conds.isEmpty()) {
+                break;
+            }
+            usedCondCount += conds.size();
+            if (conds.size() == 1) {
+                Relation r = (Relation) conds.get(0);
+                if (r.relation() == Condition.RelationType.EQ) {
+                    equalCondCount++;
+                    Object value = r.value();
+                    if (NumericUtil.isNumber(value) || value instanceof Date) {
+                        value = LongEncoding.encodeNumber(value);
+                    }
+                    prefixes.add(value);
+                    continue;
+                }
+            }
```

Comment:
update process to:

1. get the prefixes (the number must be > 1)
2. get range condition (the number must be 1 or 0)